data-prepper-plugins/s3-source/src/main/java/org/opensearch/dataprepper/plugins/source/parser/ParsedMessage.java

```diff
@@ -0,0 +1,77 @@
+package org.opensearch.dataprepper.plugins.source.parser;
+
+import org.joda.time.DateTime;
+import org.opensearch.dataprepper.plugins.source.S3EventBridgeNotification;
+import org.opensearch.dataprepper.plugins.source.S3EventNotification;
+import software.amazon.awssdk.services.sqs.model.Message;
+
+import java.util.List;
+
+public class ParsedMessage {
+    private final Message message;
+    private final boolean failedParsing;
+    private String bucketName;
+    private String objectKey;
+    private String eventName;
+    private DateTime eventTime;
+    private final int recordSize;
+    private String detailType;
+
+    public ParsedMessage(final Message message, final boolean failedParsing) {
+        this.message = message;
+        this.failedParsing = failedParsing;
+        this.recordSize = 0;
+    }
+
+    ParsedMessage(final Message message, final List<S3EventNotification.S3EventNotificationRecord> notificationRecords) {
```

Comment:
Some comments here would be nice to explain why we only looks at the first element in the list but use a recordSize equal to the list size.

Reply:
Yeh, it's usually contains only one element for our use-case. Maybe the other event types produce multiple records in same message. I can add a comment here